src/dcfldd.c

```diff
@@ -26,6 +26,8 @@
 
 /* GNU dd originally written by Paul Rubin, David MacKenzie, and Stuart Kemp. */
 
+#define _POSIX_C_SOURCE 200809L /* e.g. for pclose(3) */
```

Comment:
Consider using `AC_USE_SYSTEM_EXTENSIONS` instead.

Reply:
@thesamesam interesting pointer, thanks for making me aware. @davidpolverari how do you feel about use of `AC_USE_SYSTEM_EXTENSIONS`?

Reply:
@thesamesam it doesn't seem to work, it's doing a lot less than advertised in practice, not yet sure why. here's the difference in configure output:

```diff
--- BEFORE.txt  2024-05-05 19:40:06.598606763 +0200
+++ AFTER.txt   2024-05-05 19:40:34.662903013 +0200
@@ -29,3 +29,2 @@
 checking for egrep... /bin/grep -E
-checking for an ANSI C-conforming const... yes
 checking for stdio.h... yes
@@ -39,2 +38,7 @@
 checking for unistd.h... yes
+checking for wchar.h... yes
+checking for minix/config.h... no
+checking whether it is safe to define __EXTENSIONS__... yes
+checking whether _XOPEN_SOURCE should be defined... no
+checking for an ANSI C-conforming const... yes
 checking whether byte ordering is bigendian... no
@@ -46,3 +50,3 @@
 checking whether strtoumax is declared... yes
-checking whether strndup is declared... no
+checking whether strndup is declared... yes
 checking that generated files are newer than configure... done
```

I inserted `AC_USE_SYSTEM_EXTENSIONS` right after `AC_PROG_EGREP`.

And the compile errors are (most or even) all those that I was previously fixing:

```console
# make -k |& grep error:
dcfldd.c:253:2: error: call to undeclared function 'pclose'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
dcfldd.c:281:5: error: call to undeclared function 'kill'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
dcfldd.c:482:37: error: call to undeclared function 'popen'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
dcfldd.c:482:35: error: incompatible integer to pointer conversion assigning to 'FILE *' (aka 'struct _IO_FILE *') from 'int' [-Wint-conversion]
dcfldd.c:496:45: error: call to undeclared function 'popen'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
dcfldd.c:496:43: error: incompatible integer to pointer conversion assigning to 'FILE *' (aka 'struct _IO_FILE *') from 'int' [-Wint-conversion]
dcfldd.c:506:46: error: call to undeclared function 'popen'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
dcfldd.c:506:44: error: incompatible integer to pointer conversion assigning to 'FILE *' (aka 'struct _IO_FILE *') from 'int' [-Wint-conversion]
dcfldd.c:516:48: error: call to undeclared function 'popen'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
dcfldd.c:516:46: error: incompatible integer to pointer conversion assigning to 'FILE *' (aka 'struct _IO_FILE *') from 'int' [-Wint-conversion]
dcfldd.c:526:48: error: call to undeclared function 'popen'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
dcfldd.c:526:46: error: incompatible integer to pointer conversion assigning to 'FILE *' (aka 'struct _IO_FILE *') from 'int' [-Wint-conversion]
dcfldd.c:536:48: error: call to undeclared function 'popen'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
dcfldd.c:536:46: error: incompatible integer to pointer conversion assigning to 'FILE *' (aka 'struct _IO_FILE *') from 'int' [-Wint-conversion]
dcfldd.c:545:39: error: call to undeclared function 'popen'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
dcfldd.c:545:37: error: incompatible integer to pointer conversion assigning to 'FILE *' (aka 'struct _IO_FILE *') from 'int' [-Wint-conversion]
dcfldd.c:553:35: error: call to undeclared function 'popen'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
dcfldd.c:553:33: error: incompatible integer to pointer conversion assigning to 'FILE *' (aka 'struct _IO_FILE *') from 'int' [-Wint-conversion]
sha2.c:89:2: error: Define BYTE_ORDER to be equal to either LITTLE_ENDIAN or BIG_ENDIAN
argmatch.c:106:7: error: call to undeclared function 'strncasecmp'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
util.c:391:9: error: call to undeclared function 'fdopen'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
util.c:391:7: error: incompatible integer to pointer conversion assigning to 'FILE *' (aka 'struct _IO_FILE *') from 'int' [-Wint-conversion]
util.c:394:9: error: call to undeclared function 'fdopen'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
util.c:394:7: error: incompatible integer to pointer conversion assigning to 'FILE *' (aka 'struct _IO_FILE *') from 'int' [-Wint-conversion]
output.c:97:14: error: call to undeclared function 'popen'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
output.c:97:12: error: incompatible integer to pointer conversion assigning to 'FILE *' (aka 'struct _IO_FILE *') from 'int' [-Wint-conversion]
output.c:134:24: error: call to undeclared function 'fileno'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
```

Reply:
Make sure to include config.h everywhere?

Reply:
> Consider using `AC_USE_SYSTEM_EXTENSIONS` instead.

I see no problems using it, as we target POSIX-y OSes and we're not thinking about moving away from autotools, so it LGTM.

Reply:
@davidpolverari @thesamesam alright, let me try the `config.h` route in a minute…

Reply:
> Make sure to include config.h everywhere (and before any other includes)?

@thesamesam seems to work, pushed, thanks for the idea!